ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -103,28 +104,19 @@ public PendingTransactions(
   }
 
   public void evictOldTransactions() {
-    synchronized (pendingTransactions) {
-      final Instant removeTransactionsBefore =
-          clock.instant().minus(maxTransactionRetentionHours, ChronoUnit.HOURS);
-      final List<TransactionInfo> transactionsToRemove =
-          prioritizedTransactions.stream()
-              .filter(
-                  transaction -> transaction.getAddedToPoolAt().isBefore(removeTransactionsBefore))
-              .collect(toList());
-      transactionsToRemove.forEach(transaction -> removeTransaction(transaction.getTransaction()));
-    }
+    final Instant removeTransactionsBefore =
+        clock.instant().minus(maxTransactionRetentionHours, ChronoUnit.HOURS);
+
+    pendingTransactions.values().stream()
+        .filter(transaction -> transaction.getAddedToPoolAt().isBefore(removeTransactionsBefore))
+        .forEach(transaction -> removeTransaction(transaction.getTransaction()));
```

Comment:
It would be nice if there was some way to assert that the underlying spliterator had the CONCURRENT characteristic before performing.  While this is correct (because it is from a ConcurrentHashMap) I am concerned that future alterations may miss this nuance.

Reply:
If the map ever gets changed back to a non-concurrent map, all the places it's accessed from will be problematic.  This is actually the place that kind of mistake will be most obvious because the unit tests will fail with `ConcurrentModificationException`.  The others will just be subtle bugs in the face of multiple threads.